low-level/framework/rabbitmq/rabbitmq_egress_processor.py

```diff
@@ -255,40 +192,31 @@ def _transmit_msg_on_exchange(self):
               (self._jsonMsg.get("message").get("actuator_response_type").get("ack") is not None or \
                 self._jsonMsg.get("message").get("actuator_response_type").get("thread_controller") is not None):
                 self._add_signature()
-                jsonMsg = json.dumps(self._jsonMsg).encode('utf8')
-                self._ack_connection.publish(exchange=self._exchange_name,
-                                             routing_key=self._ack_routing_key,
-                                             properties=msg_props,
-                                             body=jsonMsg)
+                self._producer.send([json.dumps(self._jsonMsg)])
                 logger.debug("_transmit_msg_on_exchange, Successfully Sent: %s" % jsonMsg)
 
             # Routing requests for IEM msgs sent from the LoggingMsgHandler
             elif self._jsonMsg.get("message").get("IEM_routing") is not None:
                 log_msg = self._jsonMsg.get("message").get("IEM_routing").get("log_msg")
                 if self._iem_route_addr != "":
-                    self._iem_connection.publish(exchange=self._iem_route_exchange_name,
-                                                 routing_key=self._routing_key,
-                                                 properties=msg_props,
-                                                 body=str(log_msg))
+                    self._producer.send([json.dumps(self._jsonMsg)])
                 else:
                     logger.warn("RabbitMQegressProcessor, Attempted to route IEM without a valid 'iem_route_addr' set.")
                 logger.debug("_transmit_msg_on_exchange, Successfully Sent: %s" % log_msg)
             else:
                 self._add_signature()
-                jsonMsg = json.dumps(self._jsonMsg).encode('utf8')
```

Comment:
dumping with utf8 encoding keeps it generic, why removing ? producer doesnt accept utf8 encoded messages?

Reply:
Encoding is happening in message bus itself. So, no need to encode here

Reply:
Ok. also these json direct apis needs to be avoided. Need to plan to use kvstore apis, which has ability to dump in different data formats including json